pkg/paths/dirname.go

```diff
@@ -0,0 +1,32 @@
+// Package paths contains utility functions for handling file paths independently from the local OS unlike the path/filepath package.
+package paths
+
+import (
+	"regexp"
+	"strings"
+)
+
+var multipleSlashesRegex = regexp.MustCompile("/{2,}")
+
+// Dir is like filepath.Dir but it doesn't use the local OS path separator.
+func Dir(path string) string {
+	// Normalize path by replacing multiple slashes with a single slash
+	normalizedPath := multipleSlashesRegex.ReplaceAllString(path, "/")
+
+	// Your existing logic here
+	if normalizedPath == "/" {
+		return "/"
+	}
+	if strings.HasSuffix(normalizedPath, "/") {
+		return strings.TrimSuffix(normalizedPath, "/")
+	}
+	idx := strings.LastIndex(normalizedPath, "/")
+	switch idx {
+	case 0:
+		return "/"
+	case -1:
+		return "."
+	default:
+		return normalizedPath[:idx]
+	}
+}
```

Comment:
I wonder if this could just be sth. like that:

```go
import (
	"path"
	"path/filepath"
)

func Dir(pathStr string) string {
	return path.Dir(filepath.ToSlash(pathStr))
}
```

Reply:
filepath.ToSlash uses `os.PathSeparator`.

Reply:
It could use `path.Dir` as is maybe 🤔

Reply:
> filepath.ToSlash uses `os.PathSeparator`.

Seems fine to me. Would mean this is a no-op on unix-like OSes.

> It could use `path.Dir` as is maybe 🤔

OTOH, yes, do we even need the normalization at all?

Reply:
No, the input comes from static slash-separated strings.